cmd/snap-repair/main.go

```diff
@@ -59,6 +59,11 @@ func init() {
 var errOnClassic = fmt.Errorf("cannot use snap-repair on a classic system")
 
 func main() {
+	if err := snapdtool.MaybeSetupFIPS(); err != nil {
+		fmt.Fprintf(os.Stderr, "cannot check or enable FIPS mode: %v", err)
+		os.Exit(1)
```

Comment:
This should not fail on missing "snapd" symlink. In other words, repair should not depend on snapd snap being mounted and operational, as that's one of the things repair may be necessary to fix in the field.

Reply:
but I think maybe dynamic linking already fails in the new snapd snap if the symlink is not there? (cc @valentindavid @bboozzoo ) that why we added more code in various places to make sure that the symlink is there at all times, but then it might be the case though that we need to add a failure handler to the snap-repair service that tries to put the symlink in place in as much cases as possible?

Reply:
I answered the first question on my own:

```
$ ldd /snap/snapd/current/usr/lib/snapd/snap-failure 
	linux-vdso.so.1 (0x00007ffe213cd000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x0000703de6800000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x0000703de6a77000)
	/snap/snapd/current/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2 => /lib64/ld-linux-x86-64.so.2 (0x0000703de6a91000)
```

Reply:
I'll revert this change and leave a TODO instead which we can address once the uncertainty around snap-repair is resolved.

Reply:
this is now done, I left a TODO